src/tools/rbd/action/Kernel.cc

```diff
@@ -163,6 +163,7 @@ static int parse_unmap_options(char *options)
 
 static int do_kernel_showmapped(Formatter *f)
 {
+#if defined(__linux__)
```

Comment:
probably you can just ignore `Kernel.cc` and `Nbd.cc` in `rbd_srcs` in `src/tools/rbd/CMakeLists.txt` if `!WITH_KRBD` .